python-checks/src/main/java/org/sonar/python/checks/RobustCipherAlgorithmCheck.java

```diff
@@ -63,14 +86,54 @@ public class RobustCipherAlgorithmCheck extends PythonSubscriptionCheck {
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, subscriptionContext -> {
       CallExpression callExpr = (CallExpression) subscriptionContext.syntaxNode();
-      Symbol calleeSymbol = callExpr.calleeSymbol();
-      if (calleeSymbol != null) {
-        String fqn = calleeSymbol.fullyQualifiedName();
-        if (fqn != null && sensitiveCalleeFqns.contains(fqn)) {
-          subscriptionContext.addIssue(callExpr.callee(), MESSAGE);
-        }
-      }
+      Optional.ofNullable(callExpr)
+        .map(CallExpression::calleeSymbol)
+        .map(Symbol::fullyQualifiedName)
+        .filter(fqn -> sensitiveCalleeFqns.contains(fqn) || SSL_SET_CIPHERS_FQN.equals(fqn))
+        .ifPresent(str -> addMethodSpecificIssue(subscriptionContext, callExpr, str));
     });
   }
 
+  private static void addMethodSpecificIssue(
+    SubscriptionContext subscriptionContext,
+    CallExpression callExpression,
+    String fullyQualifiedName
+  ) {
+    if (!SSL_SET_CIPHERS_FQN.equals(fullyQualifiedName) || argumentSpecifiesRiskyAlgorithm(callExpression)) {
+      subscriptionContext.addIssue(callExpression.callee(), MESSAGE);
+    }
+  }
+
+  private static boolean argumentSpecifiesRiskyAlgorithm(CallExpression callExpression) {
```

Comment:
It is a good practice to name functions that returns booleans with is or has. for example `hasArgumentWithSensitiveAlgorithm`.

Reply:
You are correct. Made the change in e5c69678b8e41c64fdde93c1c58a30377d138350.